ludwig/datasets/mixins/process.py

```diff
@@ -26,7 +30,13 @@ class IdentityProcessMixin:
     processed_dataset_path: str
 
     def process_downloaded_dataset(self):
-        os.rename(self.raw_dataset_path, self.processed_dataset_path)
+        protocol, _ = split_protocol(self.processed_dataset_path)
```

Comment:
I would move this into `fs_utils` as a utility function, like `fs_utils.rename`.